src/ethereum_test_tools/vm/opcode.py

```diff
@@ -30,7 +30,51 @@ def _get_int_size(n: int) -> int:
 _push_opcodes_byte_list = [bytes([0x5F + x]) for x in range(33)]
 
 
-class Opcode(bytes):
+class OpcodeMacroBase(bytes):
+    """
+    Base class for Macro and Opcode, inherits from bytes.
+
+    This class is designed to represent a base structure for individual evm opcodes
+    and opcode macros.
+    """
+
+    _name_: str
+
+    def __new__(cls, *args):
+        """
+        Since OpcodeMacroBase is never instantiated directly but through
+        subclassing, this method simply forwards the arguments to the
+        bytes constructor.
+        """
+        return super().__new__(cls, *args)
+
+    def __call__(self, *args_t: Union[int, bytes, str, "Opcode", FixedSizeBytes]) -> bytes:
+        """
+        Make OpcodeMacroBase callable, so that arguments can directly be
+        provided to an Opcode in order to more conveniently generate
+        bytecode (implemented in the subclass).
+        """
+        # ignore opcode arguments
+        args_t.count
+        return bytes(self)
+
+    def __str__(self) -> str:
+        """
+        Return the name of the opcode, assigned at Enum creation.
+        """
+        return self._name_
+
+    def __eq__(self, other):
+        """
+        Required to differentiate between SELFDESTRUCT and SENDALL type of cases
+        And to register the Macro opcodes which are defined from same bytes
+        """
+        if isinstance(other, OpcodeMacroBase):
+            return self._name_ == other._name_
+        return NotImplemented
```

Comment:
@marioevz  if I raise NotImplementedError here it breaks many places in unit tests.

Reply:
84e3072e693516f34e5021c82906a0bcb1ff46ba returns False instead of raising or returning `NotImplementedError`

Reply:
So the fail wasn't in existing tests (tox is green), it's in the new tests in @winsvega's  `eip1153` branch. It's due to a subtle (previously unexpected?) use of `Opcode` syntax: 

https://github.com/ethereum/execution-spec-tests/blob/4e41d537664d085e4432e246e8fc7bdecaeba0f9/tests/cancun/eip1153_tstore/test_tload_reentrancy.py#L32

Here, **calling** the `Op.ADDRESS` enum (note the brackets) causes it to return the bytecode (a `bytes` object), instead of just creating an instance of `Op.ADDRESS` which is of type `Opcodes`. 

This is an easy trap to fall into, so I think this could be a good solution:
```python
    def __eq__(self, other):
        """
        Required to differentiate between SELFDESTRUCT and SENDALL type of cases
        And to register the Macro opcodes which are defined from same bytes
        """
        if isinstance(other, OpcodeMacroBase):
            return self._name_ == other._name_
        if isinstance(other, bytes):
            bytes(self) == other
        return NotImplementedError(f"Unsupported type for comparison f{type(other)}")
```

Reply:
I explicitly stated that call_dest is bytes variable. why did it compare still?